smart-campus-service-impl/src/main/java/hu/unideb/smartcampus/service/ejabberd/SharedRosterServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+      LOGGER.info("{} user added to group {}", user, group);
+    } else {
+      LOGGER.info("Adding user to group failed... Try again later.");
+    }
+  }
+
+  /**
+   * {@inheritDoc}.
+   */
+  @Override
+  public void createNewGroup(String group, String groupName, String description,
+      List<String> displayedGroups) {
+    LOGGER.info("Creating group called {}.", group);
+    WebTarget target = createClientByUrl(SHARED_ROSTER_GROUP_CREATE_COMMAND);
+    Entity<CreateGroupRequest> entity = Entity.entity(
+        CreateGroupRequest.builder().host(DOMAIN).group(group).description(description)
+            .name(groupName).display(getDisplayedGroups(displayedGroups)).build(),
+        MediaType.APPLICATION_JSON);
+    Response post = target.request().post(entity);
+    if (isSuccessful(post)) {
+      LOGGER.info("{} created.", group);
+    } else {
+      LOGGER.info("Creating group failed... Try again later.");
+    }
+  }
+
+  /**
+   * {@inheritDoc}.
+   */
+  @Override
+  public void deleteUserFromGroup(String user, String group) {
+    LOGGER.info("Deleting user {} from group called {}.", user, group);
+    WebTarget client = createClientByUrl(SHARED_ROSTER_GROUP_DELETE_USER_COMMAND);
+    Entity<DeleteUseRequest> entity =
+        Entity.entity(DeleteUseRequest.builder().host(DOMAIN).group(group).user(user).build(),
+            MediaType.APPLICATION_JSON);
+    Response post = client.request().post(entity);
+    if (isSuccessful(post)) {
+      LOGGER.info("{} deleted from group {}.", user, group);
+    } else {
+      LOGGER.info("Deleting user failed... Try again later.");
+    }
+  }
+
+  /**
+   * {@inheritDoc}.
+   */
+  @Override
+  public Map<String, String> getGroupInformation(String group) {
+    LOGGER.info("Getting group information, {}.", group);
+    WebTarget client = createClientByUrl(SHARED_ROSTER_GROUP_INFO_COMMAND);
+    Entity<InformationRequest> entity = Entity.entity(
+        InformationRequest.builder().host(DOMAIN).group(group).build(), MediaType.APPLICATION_JSON);
+    return client.request().post(entity).readEntity(new GenericType<Map<String, String>>() {});
+  }
+
+  /**
+   * {@inheritDoc}.
+   */
+  @Override
+  public List<String> getGroupList() {
+    LOGGER.info("Getting group list.");
+    WebTarget target = createClientByUrl(SHARED_ROSTER_GROUP_LIST_COMMAND);
+    Entity<GroupRequest> entity =
+        Entity.entity(new GroupRequest(DOMAIN), MediaType.APPLICATION_JSON);
+    return target.request().post(entity).readEntity(new GenericType<List<String>>() {});
+  }
+
+  private WebTarget createClientByUrl(String url) {
+    Client client = ClientBuilder.newClient();
+    return client.target(HOST + url);
+  }
+
+  private String getDisplayedGroups(List<String> displayedGroups) {
+    StringBuilder builder = new StringBuilder();
+    displayedGroups.stream().forEach(p -> builder.append(p).append("\n"));
+    return builder.toString();
+  }
+
+  private boolean isSuccessful(Response post) {
```

Comment:
Could we create a new service(HttpStatusValidator) which can check the Response object?
At present, just one method: isOk(response)  ->> MediaType.OK (maybe it is better than the magic numbers)